lightningd/dual_open_control.c

```diff
@@ -645,13 +645,23 @@ openchannel2_hook_cb(struct openchannel2_payload *payload STEALS)
 		return subd_send_msg(dualopend, take(msg));
 	}
 
+	// Determine the wallet index for our upfront_shutdown_script, UINT32_MAX if not found.
+	u32 our_shutdown_script_wallet_index = UINT32_MAX;
+	bool is_p2sh;
+	wallet_can_spend(
```

Comment:
here and elsewhere, shouldn't we do something special if `wallet_can_spend()` returns false?

Reply:
Well, actually it's fine as it is, because it's not an error if it's not in the wallet - it may be whitelisted.